src/group_theory/subgroup.lean

```diff
@@ -1308,6 +1308,18 @@ begin
   { exact λ h, le_bot_iff.mp (λ x hx, h (hx.trans f.map_one.symm)) },
 end
 
+@[to_additive]
+lemma ker_prod_map {G' : Type*} {N' : Type*} [group G'] [group N'] (f : G →* N) (g : G' →* N') :
+  (prod_map f g).ker = prod f.ker g.ker :=
+begin
+  ext x,
+  refine ⟨λ h, _, λ h, _⟩,
+  { rw mem_ker at h,
+    simpa [mem_prod, mem_ker] using h },
+  { rw [mem_prod, mem_ker, mem_ker] at h,
+    simpa [mem_ker] using h }
+end
```

Comment:
I think a nicer way to prove this is:

```suggestion
@[to_additive]
lemma prod_comap_prod_map {G' : Type*} {N' : Type*} [group G'] [group N']
  (f : G →* N) (g : G' →* N') (S : subgroup N) (S' : subgroup N') :
  (S.prod S').comap (prod_map f g) = prod (S.comap f) (S'.comap g) :=
set_like.coe_injective $ prod.map_preimage_set_prod f g _ _

@[to_additive]
lemma ker_prod_map {G' : Type*} {N' : Type*} [group G'] [group N'] (f : G →* N) (g : G' →* N') :
  (prod_map f g).ker = prod f.ker g.ker :=
begin
  dsimp only [ker],
  rw [←prod_comap_prod_map _ _ _ _, bot_prod_bot],
end
```
which needs this lemma, probably next to either `prod.map` or `set.prod`:
```lean
lemma prod.map_preimage_set_prod {α β γ δ : Type*}
  (f : α → γ) (g : β → δ) (c : set γ) (d : set δ) :
  prod.map f g ⁻¹' c.prod d = (f ⁻¹' c).prod (g ⁻¹' d) :=
rfl
```